exporter/internal/queue/batcher.go

```diff
@@ -0,0 +1,259 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package queue // import "go.opentelemetry.io/collector/exporter/internal/queue"
+
+import (
+	"context"
+	"math"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter/exporterbatcher"
+	"go.opentelemetry.io/collector/exporter/internal"
+)
+
+type batch struct {
+	ctx     context.Context
+	req     internal.Request
+	idxList []uint64
+}
+
+type Batcher struct {
```

Comment:
Do we plan to soon delete BatchSender. I expect that to be the case after we have a "blocking 1 element queue"?

Reply:
Do you see that we may need multiple implementations of the Batcher? If yes, we can probably use an interface?

Reply:
I am leaning toward getting rid of `batch_sender` and implementing the blocking logic in as a blocking queue ([like this](https://github.com/open-telemetry/opentelemetry-collector/pull/11447/files#diff-cc61518e1acc5a429a7fc835d303a127d1f3f5160e2df552dd38766e6e1a9fc9)). That way `queue_sender` (or maybe we should call it `queue_batch_sender`) would handle all cases so we don't maintain two separate senders. I do find it slightly awkward that the blocking queue is holding `map[int]chan` in order to inform blocking senders about completion though. WDYT?